kotlin/tests/src/test/kotlin/com/squareup/moshi/kotlin/DualKotlinTest.kt

```diff
@@ -297,6 +298,13 @@ class DualKotlinTest(useReflection: Boolean) {
       """{"i":6}"""
     val result = adapter.fromJson(testJson)!!
     assertThat(result.i).isEqualTo(6)
+
+    // TODO doesn't work yet.
```

Comment:
Is this https://github.com/square/moshi/issues/1170 ?

Reply:
yeah. Supporting value classes only "works" as long as you don't have a default parameter value, but we'd need to add something to recognize that

Reply:
linked the issue in the comment 4786053